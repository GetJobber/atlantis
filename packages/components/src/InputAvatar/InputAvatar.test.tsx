import React from "react";
import { cleanup, fireEvent, render, waitFor } from "@testing-library/react";
import { InputAvatar } from ".";

afterEach(cleanup);

const testFile = new File(["ðŸ”± Atlantis"], "atlantis.png", {
  type: "image/png",
});

function fetchUploadParams(file: File) {
  return Promise.resolve({
    key: file.name,
    url: "https://httpbin.org/post",
    fields: { secret: "ðŸ¤«" },
  });
}

it("renders", () => {
  const { container } = render(
    <InputAvatar getUploadParams={fetchUploadParams} />,
  );
  expect(container).toMatchSnapshot();
});

it("renders with a provided image", () => {
  const { container } = render(
    <InputAvatar
      getUploadParams={fetchUploadParams}
      imageUrl="https://api.adorable.io/avatars/150/jobbler"
    />,
  );
  expect(container).toMatchSnapshot();
});

it("properly notifies upload callbacks", async () => {
  const changeHandler = jest.fn();
  const completionHandler = jest.fn();
  const { container } = render(
    <InputAvatar
      getUploadParams={fetchUploadParams}
      onChange={changeHandler}
      onUploadComplete={completionHandler}
    />,
  );

  const input = container.querySelector("input[type=file]");

  fireEvent.change(input, { target: { files: [testFile] } });

  await waitFor(() => {
    expect(changeHandler).toHaveBeenCalledTimes(1);
    expect(changeHandler).toHaveBeenCalledWith({
      key: "atlantis.png",
      name: "atlantis.png",
      size: expect.any(Number),
      progress: expect.any(Number),
      src: expect.any(Function),
      type: "image/png",
    });

    expect(completionHandler).toHaveBeenCalledTimes(1);
    expect(completionHandler).toHaveBeenCalledWith({
      key: "atlantis.png",
      name: "atlantis.png",
      size: expect.any(Number),
      progress: 1,
      src: expect.any(Function),
      type: "image/png",
    });
  });
});

it("should allow for avatar removal", async () => {
  const changeHandler = jest.fn();
  const { getByText } = render(
    <InputAvatar
      getUploadParams={fetchUploadParams}
      onChange={changeHandler}
      imageUrl="https://api.adorable.io/avatars/150/jobbler"
    />,
  );

  fireEvent.click(getByText("Remove"));

  await waitFor(() => {
    expect(changeHandler).toHaveBeenCalledTimes(1);
    expect(changeHandler).toHaveBeenCalledWith(undefined);
  });
});
