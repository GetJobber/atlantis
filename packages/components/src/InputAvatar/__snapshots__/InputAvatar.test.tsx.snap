// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div>
  <div
    class="inputAvatar"
  >
    <div
      class="preview"
    >
      <div
        class="avatar large isDark"
      >
        <svg
          class="ZDlJU6tECg7ouG-b27gya _2SoAwA0A8G1BGDNjtkt-lt"
          data-testid="person"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            class="_2GsLyQLHv8yNULHeXGdver _29W47x5R7CHDPdmq99tUBc"
            d="M512 469.333c-94.255 0-170.665-76.41-170.665-170.667s76.41-170.667 170.665-170.667c94.259 0 170.667 76.41 170.667 170.667s-76.407 170.667-170.667 170.667zM512 384c47.13 0 85.333-38.205 85.333-85.333s-38.204-85.333-85.333-85.333c-47.125 0-85.333 38.205-85.333 85.333s38.208 85.333 85.333 85.333z"
          />
          <path
            class="_2GsLyQLHv8yNULHeXGdver _29W47x5R7CHDPdmq99tUBc"
            d="M225.679 810.667h572.644c-36.698-123.392-151.1-213.333-286.323-213.333s-249.623 89.941-286.321 213.333zM810.795 810.654l-0.115 0.013zM213.322 810.667l-0.114-0.013zM129.919 811.332c38.557-171.341 199.096-299.332 382.081-299.332 182.989 0 343.629 127.991 382.187 299.332 1.263 5.615 1.788 11.123 1.647 16.461-0.068 2.722-0.311 5.402-0.717 8.026-5.303 34.441-38.46 60.181-76.902 60.181h-612.327c-39.024 0-72.603-26.526-77.123-61.756-0.939-7.313-0.624-15.006 1.155-22.912z"
          />
        </svg>
      </div>
    </div>
    <div
      class="dropZoneBase"
      tabindex="-1"
    >
      <input
        accept="image/*"
        autocomplete="off"
        style="display: none;"
        tabindex="-1"
        type="file"
      />
      <button
        class="button small work secondary fullWidth"
        type="button"
      >
        <span
          class="base extraBold smaller uppercase green"
        >
          Upload Image
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`renders with a provided image 1`] = `
<div>
  <div
    class="inputAvatar"
  >
    <div
      class="preview"
    >
      <div
        class="avatar large isDark"
        role="img"
        style="background-image: url(https://api.adorable.io/avatars/150/jobbler);"
      />
    </div>
    <div
      class="dropZoneBase"
      tabindex="-1"
    >
      <input
        accept="image/*"
        autocomplete="off"
        style="display: none;"
        tabindex="-1"
        type="file"
      />
      <button
        class="button small work secondary fullWidth"
        type="button"
      >
        <span
          class="base extraBold smaller uppercase green"
        >
          Change Image
        </span>
      </button>
    </div>
    <button
      class="button small destructive secondary"
      type="button"
    >
      <span
        class="base extraBold smaller uppercase red"
      >
        Remove
      </span>
    </button>
  </div>
</div>
`;
